nautobot/utilities/filters.py

```diff
@@ -369,66 +373,91 @@ def _get_filter_lookup_dict(existing_filter):
-            field = get_model_field(cls._meta.model, field_name)
+            magic_filters[new_filter_name] = new_filter
 
-            # Create new filters for each lookup expression in the map
-            for lookup_name, lookup_expr in lookup_map.items():
-                new_filter_name = "{}__{}".format(existing_filter_name, lookup_name)
+        return magic_filters
 
-                try:
-                    if existing_filter_name in cls.declared_filters:
-                        # The filter field has been explicity defined on the filterset class so we must manually
-                        # create the new filter with the same type because there is no guarantee the defined type
-                        # is the same as the default type for the field
-                        resolve_field(field, lookup_expr)  # Will raise FieldLookupError if the lookup is invalid
-                        new_filter = type(existing_filter)(
-                            field_name=field_name,
-                            lookup_expr=lookup_expr,
-                            label=existing_filter.label,
-                            exclude=existing_filter.exclude,
-                            distinct=existing_filter.distinct,
-                            **existing_filter.extra,
-                        )
-                    else:
-                        # The filter field is listed in Meta.fields so we can safely rely on default behaviour
-                        # Will raise FieldLookupError if the lookup is invalid
-                        new_filter = cls.filter_for_field(field, field_name, lookup_expr)
-                except django_filters.exceptions.FieldLookupError:
-                    # The filter could not be created because the lookup expression is not supported on the field
-                    continue
+    @classmethod
+    def add_filter(cls, new_filter_name, new_filter_field):
+        """
+        Allow filters to be added post-generation on import.
+
+        Will provide `<field_name>__<lookup_expr>` generation automagically.
+        """
+        if new_filter_name in cls.base_filters:
+            logger.error("There was a conflict with filter `%s`, the custom filter was ignored." % new_filter_name)
+
+        if not isinstance(new_filter_field, django_filters.Filter):
+            raise TypeError(f"Tried to add filter ({new_filter_name}) which is not an instance of Django Filter")
 
-                if lookup_name.startswith("n"):
-                    # This is a negation filter which requires a queryset.exclude() clause
-                    # Of course setting the negation of the existing filter's exclude attribute handles both cases
-                    new_filter.exclude = not existing_filter.exclude
+        cls.base_filters[new_filter_name] = new_filter_field
+        cls.base_filters.update(
+            cls._generate_lookup_expression_filters(filter_name=new_filter_name, filter_field=new_filter_field)
```

Comment:
I like the idea here - does it work correctly for all three filter fields added by the example-plugin? Would be awesome to have a test case or two confirming this.

Reply:
It should all automagically work.

Reply:
As called out here: https://github.com/nautobot/nautobot/pull/1496/files#r845713945

Calling `get_filters` no longer gives a full picture of the filters, since the plugins have to add them after generation,

If we need `get_filters` to generate the same list as `base_filters` then we can create a registry in the class for "plugged-in" fields and in our `get_filters` method merge that with the list we get from calling `super().get_filters()`

Reply:
TODO: Add test coverage to `_generate_lookup_expression_filters`, `add_filter`, `get_filters`